From c0cec3409f3abda1e2359a79ccac575b4ea1838b Mon Sep 17 00:00:00 2001
From: Gary Lin <glin@suse.com>
Date: Tue, 21 May 2019 16:56:06 +0800
Subject: [PATCH 1/1] OvmfPkg/EnrollDefaultKeys: Set the fixed time

For the reproducible build, we need to set the fixed time when setting
the authenticate variables.

Signed-off-by: Gary Lin <glin@suse.com>
---
 OvmfPkg/EnrollDefaultKeys/EnrollDefaultKeys.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

Index: edk2-edk2-stable202202/OvmfPkg/EnrollDefaultKeys/EnrollDefaultKeys.c
===================================================================
--- edk2-edk2-stable202202.orig/OvmfPkg/EnrollDefaultKeys/EnrollDefaultKeys.c
+++ edk2-edk2-stable202202/OvmfPkg/EnrollDefaultKeys/EnrollDefaultKeys.c
@@ -324,6 +324,15 @@ EnrollListOfCerts (
     goto FreeData;
   }
 
+  // Set the fixed time for the reproducible build
+  // 2019-5-20 00:00:00
+  SingleHeader->TimeStamp.Year       = 2019;
+  SingleHeader->TimeStamp.Month      = 5;
+  SingleHeader->TimeStamp.Day        = 20;
+  SingleHeader->TimeStamp.Hour       = 0;
+  SingleHeader->TimeStamp.Minute     = 0;
+  SingleHeader->TimeStamp.Second     = 0;
+ 
   SingleHeader->TimeStamp.Pad1       = 0;
   SingleHeader->TimeStamp.Nanosecond = 0;
   SingleHeader->TimeStamp.TimeZone   = 0;
